### PR TITLE
[Create WS] Various fixes

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Several smaller improvements to the create workspace page
- apply the same preprocessing of the given contextURL for resolving a workspace context that is applied when creating a workspace.
- handle `referrer` prefix URLs, so that the workspace starts immediately
- ensure createWorkspace is not called multiple times
- checks and displays directly if a workspace on the same revision is already running

Screenshot 2023-04-06 at 11 55 43
cc @gtsiolis 

## Related Issue(s)

Fixes https://linear.app/gitpod/issue/WEB-68/startworkspacewithoptions-invalid-url-error-on-missing-scheme
Fixes https://linear.app/gitpod/issue/WEB-81/context-urls-with-referrer-dont-show-modal
Fixes https://linear.app/gitpod/issue/WEB-73/[new-workspace-page]-dont-start-more-than-one-workspace-with-enter

## How to test

- Try starting a workspace with a contextURL that lacks the https:// part: https://se-create-ws-page.preview.gitpod-dev.com/#github.com/gitpod-io/website
- start a workspace using referrer prefix : https://se-create-ws-page.preview.gitpod-dev.com/#referrer:jetbrains-gateway:phpstorm/https://github.com/gitpod-samples/template-php-laravel-mysql
- try starting multiple workspace by hitting enter
- try start multiple workspaces on the same revision and see how the create workspace page displays those workspaces

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (easyCZ) [inline components/server/src/workspace/gitpod-server-impl.ts:3672]: Is there a time when you wouldn't want to normalize? And if no, can we move the normalization inside the `contextParser.handle` method?

Comment (svenefftinge) [reply components/server/src/workspace/gitpod-server-impl.ts:3672]: There are places where we don't do the normalization, which is when contexts are created based on git hoster events, where we don't need to be fuzzy around the allowed input. It would probably not harm to call the normalization in those cases as well, but I would like to avoid changing the implementation of contextParser with this PR. It's a bit complex and I don't see enough value in the proposed change.

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Review by gtsiolis — COMMENTED** [review]
Took a quick look, but I couldn't open any workspace. Getting the following error.

Screenshot 2023-04-06 at 20 12 21

I'll see how we can improve the parallel workspaces listing, too.

File: components/dashboard/src/workspaces/CreateWorkspacePage.tsx
Comment (gtsiolis) [inline components/dashboard/src/workspaces/CreateWorkspacePage.tsx:262]: **thought**: I understand revision is probably more accurate, in case we could relax this a bit we could go with a friendlier and less scary term. Your call.

```suggestion
                        <p className="mt-6 text-center text-base">Running workspaces on this branch</p>
```

Comment (gtsiolis) [inline components/dashboard/src/workspaces/CreateWorkspacePage.tsx:262]: **question**: If it's not too much trouble and we want to add this in the scope of this PR, it could be helpful to be specific about the number of similar workspaces.

```
// BEFORE
Running workspaces on this branch

// AFTER
3 running workspaces on this branch
```

Comment (gtsiolis) [inline components/dashboard/src/workspaces/CreateWorkspacePage.tsx:262]: **question**: Definitely out of the scope of this PR, but I wonder if including also stopped workspaces could make sense here. If a user has a stopped workspace on the same branch (revision), does it make sense to also include it here?

Comment (gtsiolis) [inline components/dashboard/src/workspaces/CreateWorkspacePage.tsx:262]: **praise**: I can't skip saying how good this addition generally feels. ✨

Comment (gtsiolis) [inline components/dashboard/src/workspaces/CreateWorkspacePage.tsx:262]: **question**: I couldn't hold from thinking how this list could be better visualized. What do you think of collapsing the list inside a dropdown?

For running workspaces, I guess this won't be usually more than 2 or 3, but there could be edge cases where this could break the UI by overflowing the height of this feature. Additionally, replacing the primary action button with a block button is something that looks needed here.

| BEFORE | AFTER |
|-|-|
|  Welcome | Welcome-1 |

File: components/dashboard/src/workspaces/CreateWorkspacePage.tsx
Comment (gtsiolis) [inline components/dashboard/src/workspaces/CreateWorkspacePage.tsx:272]: **suggestion**: If we follow the dropdown approach, here's how this could look like:

| :a: | :b: |
|-|-|
| Welcome-1 | Welcome |

**Review by gtsiolis — APPROVED** [review]
Took another look at this and added some UX suggestions for the overall approach and copy.

Thanks for adding this, @svenefftinge. 🌊 🌊 

Approving to unblock but holding to let someone take a closer look at the code changes and possibly update the UX.

/hold

File: components/dashboard/src/workspaces/CreateWorkspacePage.tsx
Comment (svenefftinge) [reply components/dashboard/src/workspaces/CreateWorkspacePage.tsx:262]: This would not be correct. We are comparing the revision

**Pull request merged**
